Integrations/CounterCraft/CounterCraft.py

```diff
@@ -0,0 +1,873 @@
+    @return string A dict containing the response in JSON format or an error
+    """
+
+    headers = {
+        X_API_KEY_AUTHENTICATION_HEADER_PREFIX
+        + "Date": time.strftime("%a, %d %b %Y %H:%M:%S +0000", time.gmtime())
+    }
+    headers.update(DEF_HEADERS)
+    signature = get_signature(
+        request_method,
+        headers,
+        path,
+        urllib.parse.urlencode(params).encode("utf-8"),
+        SECRET_KEY,
+    )
+    headers["Authorization"] = "APIKey %s:%s" % (API_KEY, signature)
+    url = SERVER + path
+
+    res = requests.request(
+        request_method,
+        url,
+        data=json.dumps(data),
+        params=params,
+        headers=headers,
+        verify=VERIFY_CERTIFICATE,
+    )
+
+    if not res.ok:
+        demisto.debug(res.text)
+        return_error("Could not execute the request: %s" % res.text)
+
+    try:
+        res_json = res.json()
+        return res_json
+    except Exception as ex:
+        demisto.debug(str(ex))
+        return_error(str(ex))
+
+
+def return_results(title, content, human_readable, context, headers):
+    """
+    Generic function that receives a result json, and turns it into an entryObject
+    @param $title the table title
+    @param $content the object contents in JSON format
+    @param $human_readable human readable data for the table
+    @param $context the entry context
+    @param $headers table headers
+    @return dict the entryObject
+    """
+
+    if len(content) == 0:
+        return_outputs(
+            readable_output="There is no output result",
+            outputs=context,
+            raw_response=content,
+        )
+
+    if headers:
+        if isinstance(headers, str):
+            headers = headers.split(",")
+
+    else:
+        if isinstance(content, dict):
+            headers = list(set(headers).intersection(set(content.keys())))
+
+    readable_output = tableToMarkdown(
+        title,
+        human_readable,
+        headers,
+        lambda h: h.title().replace("_", " ").replace(".", ":"),
+    )
+
+    return_outputs(
+        readable_output=readable_output, outputs=context, raw_response=content
+    )
+
+
+""" COMMANDS """
+
+
+def test_module_command():
+
+    http_request("GET", "/campaigns")
+    demisto.results("ok")
+    sys.exit(0)
+
+
+def list_dsns_command():
+    """
+    Retrieve all deception support nodes
+    """
+
+    res = http_request("GET", "/deception_support_nodes")
+
+    new_dsns = []
+    for o in res["data"]:
+        new_dsns.append(
+            {
+                new_key: o[old_key] if old_key in o else None
+                for old_key, new_key in DSN_FIELDS.items()
+            }
+        )
+    context = createContext(new_dsns, removeNull=True)
+
+    return_results(
+        "Deception Support Node",
+        res["data"],
+        new_dsns,
+        {"CounterCraft.DSN(val.ID && val.ID === obj.ID)": context},
+        headers=["ID", "Name", "Description", "Hostname", "Port"],
```

Comment:
in this case, it would be great also to output to [host context standard](https://demisto.pan.dev/docs/context-standards/#host)

Reply:
Not sure about that. The DSN are special hosts that we use for both collecting and proxying purposes, I don't see a need to use the host context standard as they are different